docs/src/docs/asciidoc/user/serializers-copiers.adoc

```diff
@@ -0,0 +1,170 @@
+---
+---
+ifndef::sourcedir[]
+include::common.adoc[]
+endif::sourcedir[]
+
+= Serializers and Copiers
+
+
+ifdef::notBuildingForSite[]
+include::menu.adoc[]
+endif::notBuildingForSite[]
+
+== Intro
+
+The on-heap store is capable of storing cached objects either by reference (where the given key and value references
+are stored) or by value (where a copy of the given key and value are made and those copies are then stored) while all
+other stores are only capable of storing by value.
+
+Serializers and Copiers are the components that perform the required copies.
+
+=== Serializers
+
+All stores but the on-heap one need some form of serialization/deserialization of
+objects to be able to store and retrieve them as they cannot internally store plain java objects but only bunch-of-bytes representations
+of them. Serializers are the Ehcache mechanism that revolves this logic: every cache that has at least one store that
+cannot store by reference is going to use a pair of Serializers: one for the key and another one for the value.
+
+Serializers are scoped at the cache level: a cache can have a serializer for its keys and another one for its values.
+All stores of a cache using serializers will share a common pair of key/value ones.
```

Comment:
All stores of a cache will be using and sharing the same key and value serializer pair.